tests/integration/test_installation.py

```diff
@@ -322,3 +358,91 @@ def test_uninstallation(ws, sql_backend, new_installation):
         ws.jobs.get(job_id=assessment_job_id)
     with pytest.raises(NotFound):
         sql_backend.execute(f"show tables from hive_metastore.{install.config.inventory_database}")
+
+
+def test_fresh_global_installation(new_installation):
+    global_installation = new_installation(single_user_install=False)
+    global_installation.uninstall()
+
+
+def test_single_user_installation(new_installation):
+    single_user_installation = new_installation(single_user_install=True)
+    single_user_installation.uninstall()
+
+
+def test_global_installation_on_existing_global_install(new_installation):
+    # TODO: Finish up the initial install and then pass the prefix
+    existing_global_installation = new_installation(single_user_install=False)
+    mock_product_value = existing_global_installation.folder[-4:]
+    reinstall_global = new_installation(
+        single_user_install=False, fresh_install=False, existing_installation_prefix=mock_product_value
+    )
+    reinstall_global.uninstall()
+
+
+def test_user_installation_on_existing_global_install(new_installation):
+    # existing install at global level
+    existing_global_installation = new_installation(single_user_install=False)
+    mock_product_value = existing_global_installation.folder[-4:]
+
+    # reinstall at user level should update global install
+    reinstall_user = new_installation(
+        single_user_install=True, fresh_install=False, existing_installation_prefix=mock_product_value
+    )
+
+    # environment variable to force user install
+    os.environ['UCX_FORCE_INSTALL'] = "user"
```

Comment:
this would leak between tests, as they run in parallel, producing unreliable flaky results

Reply:
What would be the best way to test this functionality in this case? Create a mock for the environment variable so that its scope is just for that test?

Reply:
create `dict[str,str]` constructor argument